instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxSqlClientAttributesGetter.java

```diff
@@ -44,4 +44,15 @@ public String getConnectionString(VertxSqlClientRequest request) {
   public Collection<String> getRawQueryTexts(VertxSqlClientRequest request) {
     return singleton(request.getQueryText());
   }
+
+  @Nullable
+  @Override
+  public String getResponseStatusFromException(Throwable throwable) {
+    // todo
```

Comment:
@laurit can you help here as well?

Reply:
In our test the exception is of type `io.vertx.pgclient.PgException` https://github.com/eclipse-vertx/vertx-sql-client/blob/4.0.0/vertx-pg-client/src/main/java/io/vertx/pgclient/PgException.java you could extract the error from there. Probably the other vetx database drivers provide similar exceptions. You may need to use reflection to extract it.